### PR TITLE
Fix broken docs link on Downloads page

### PR DESCRIPTION
**Pull request by jdaudier**
Plus a few other misc minor verbiage updates

****

### CODE REVIEW

File: src/sections/_nav.jade [hunk 1]
```diff
@@ -10,7 +10,7 @@ header.global-nav
     nav.nav-slider.closed(role="navigation")
       ul.nav-slider__list
         li.left
-          a(itemprop="significantLink", href="docs/current") Docs
+          a(itemprop="significantLink", href="/docs/current") Docs
           span.middot ·
         li.left
           a(itemprop="significantLink", href="annotated-src/backbone.marionette.html") Annotated Source
```
Comment (samccone) [inline src/sections/_nav.jade]: we want a / here also right?

Comment (samccone) [inline src/sections/_nav.jade]: yes we do

**Comment by samccone** [comment]
good call on the copy changes @jdaudier !
and good catch with the links

**Comment by jdaudier** [comment]
ok updated `a(itemprop="significantLink", href="/docs/current") Docs`

**Comment by samccone** [comment]
dope great calls @jdaudier

**Comment by jasonLaster** [comment]
Oh shit!

**Pull request merged**